frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -439,9 +466,18 @@ export const transformTopologyData = (
               application,
             ),
           ];
-          groupsData = [
-            ...getTopologyGroupItems(deploymentConfig, topologyGraphAndNodeData.graph.groups),
-          ];
+          if (isHelmReleaseNode(deploymentConfig)) {
+            groupsData = [
+              ...getTopologyHelmReleaseGroupItem(
+                deploymentConfig,
+                topologyGraphAndNodeData.graph.groups,
+              ),
+            ];
+          } else {
+            groupsData = [
+              ...getTopologyGroupItems(deploymentConfig, topologyGraphAndNodeData.graph.groups),
+            ];
+          }
```

Comment:
It would be good to move HelmReleaseGroupItem call  into `getTopologyGroupItems` function, we could return different group based on the conditions there. So that later on we can have all group related login in one place rather that adding more if else here. Any thoughts ?

Reply:
fixed